### ISSUE TITLE
warp_frame_depth: an image and depth of different sizes are not rejected; the image is stretched onto the depth grid

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4800]
`warp_frame_depth(image_src, depth_dst, src_trans_dst, camera_matrix)` does not check that `image_src` and `depth_dst` have the same `H` and `W`. The sampling grid is normalized with the depth's size, so with a smaller depth map the whole source image gets resampled onto the depth grid. The call does not raise, and it does not warp.

With the identity transform and a 3x4 depth over a 6x8 image, you would expect either an error or the top-left 3x4 crop. What comes back is a stretched copy of the entire image:

```python
import torch
from kornia.geometry.depth import warp_frame_depth

img = torch.arange(48.0).view(1, 1, 6, 8)
out = warp_frame_depth(img, torch.ones(1, 1, 3, 4), torch.eye(4)[None], torch.eye(3)[None])
print(out[0, 0])
# [[ 0.00,  2.33,  4.67,  7.00],
#  [20.00, 22.33, 24.67, 27.00],
#  [40.00, 42.33, 44.67, 47.00]]
```

The corner values `0`, `7`, `40`, `47` are the source image's own corners. With the identity pose, pixel `(u, v)` of the destination should sample pixel `(u, v)` of the source, but it samples `(u * 7/3, v * 5/2)`.

A fix would add a `KORNIA_CHECK` that `image_src.shape[-2:] == depth_dst.shape[-2:]`. The alternative is to normalize the grid with the image's size, which gives the crop. The first is simpler, and it matches the one-frame-pair contract.

Found during the review of #4790. kornia `main` @ 57cd68e6f, torch 2.14.0, CPU.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).

**Comment by ethanstoner** [comment]
I'd like to take this one. `kornia/geometry/depth.py:427-428` normalizes the grid with `depth_dst`'s height and width, and nothing checks `image_src` against them, so the 6x8 image in the repro gets stretched onto the 3x4 grid. I'll add the `KORNIA_CHECK` you suggested and open a PR shortly.

**Comment by ducha-aiki** [comment]
@ethanstoner it's yours, and #4861 is the PR for it. We can't assign outside contributors, so this comment is the claim. No in-tree pin or prose names #4800 on `main` (`2fa713d0c`), so there is nothing else to reconcile.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
